.github/workflows/deploy-to-dev.yml

```diff
@@ -0,0 +1,167 @@
+        uses: docker/login-action@v2
+        with:
+          username: ${{ env.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PROD_TOKEN }}
+
+      - name: Build Docker Image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          file: Dockerfile-ut-func
+          load: true
+          tags: |
+            ${{ env.REPO_NAME }}:${{ steps.gen_img_version.outputs.version }}
+
+      - name: Run Tests
+        run: |
+          docker run ${{ env.REPO_NAME }}:${{ steps.gen_img_version.outputs.version }} npm run test:ci
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and Push Multi-platform Images
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          file: Dockerfile-ut-func
+          push: true
+          tags: |
+            ${{ env.REPO_NAME_OLD }}:${{ steps.gen_img_version.outputs.version }}
+            ${{ env.REPO_NAME }}:${{ steps.gen_img_version.outputs.version }}
+          platforms: |
+            linux/amd64
+            linux/arm64
+          build-args: |
+            VERSION=${{ steps.gen_img_version.outputs.version }}
+            COMMIT_HASH=${{ github.sha }}
+
+  create-pull-request:
+    name: Update Helm Charts For Production and Create Pull Request
+    runs-on: ubuntu-latest
+    needs: [ build-transformer-image, build-user-transformer-image ]
+    env:
+      VERSION: ${{ needs.build-transformer-image.outputs.version }}
+      UT_VERSION: ${{ needs.build-user-transformer-image.outputs.version }}
+    steps:
+      - name: Print Docker Image Versions
+        run: |
+          echo "Transformer: $VERSION"
+          echo "User Transformer: $UT_VERSION"
+
+      # In order to make a commit, we need to initialize a user.
+      # You may choose to write something less generic here if you want, it doesn't matter functionality wise.
+      - name: Initialize Mandatory Git Config
+        run: |
+          git config user.name "GitHub Actions"
+          git config user.email "noreply@github.com"
+
+      - name: Clone rudder-devops
+        run: |
+          git clone https://${{secrets.PAT}}@github.com/rudderlabs/rudder-devops.git
+          cd rudder-devops
+
+      - name: Update Helm Charts
+        run: |
+          cd helm-charts/shared-services/per-az
+          yq eval -i ".rudder-transformer.image.tag=\"$VERSION\"" values.dev.yaml
+          yq eval -i ".user-transformer.image.tag=\"$UT_VERSION\"" values.dev.yaml
+          git add values.dev.yaml
+
+          cd ../../config-be-rudder-transformer
+          yq eval -i ".config-be-rudder-transformer.image.tag=\"$VERSION\"" values.dev.yaml
+          yq eval -i ".config-be-user-transformer.image.tag=\"$UT_VERSION\"" values.dev.yaml
+          git add values.dev.yaml
+
+      - name: Push Changes
+        run: |
+          git commit -m "upgrade development env shared transformers"
+          git push
+
```

Comment:
Add step to auto-merge post commit push
Ref: https://github.com/marketplace/actions/merge-pull-requests-automerge-action

Reply:
Actually, we are directly pushing to the `master` branch here. Hence, no need for any pull requests.
Do you want to change this mechanism?

Reply:
Better to via PR to `develop`. In case of any release issue, first thing Infra or anyone check is the recent PR's & the corresponding versions released.
As we have auto-merge plugin, automation remain same but we can get the tracking. And some point it time, they might enforce rules on `master`

Reply:
Done.